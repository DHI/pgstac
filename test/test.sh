#!/bin/bash
tempid=$(cat /dev/urandom | tr -dc 'a-zA-Z' | fold -w 8 | head -n 1)
tempdb="pgstac_tst_${tmpid}"
DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )"


createdb $tempdb;
export PGDATABASE=$tempdb

cd $DIR/..
psql -f pgstac.sql


cat test/testdata/collections.ndjson | psql -c "copy pgstac.collections_staging FROM stdin"
cat test/testdata/items.ndjson | psql -c "copy pgstac.items_staging FROM stdin"

psql -X -f test/test.sql >/tmp/$tempdb.out

cmp --silent /tmp/$tempdb.out test/test.out && echo '### SUCCESS: Files Tests Pass! ###' || echo '### WARNING: Tests did not pass! ###' && diff /tmp/$tempdb.out test/test.out

dropdb $tempdb;
rm /tmp/$tempdb.out