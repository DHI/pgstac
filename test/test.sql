SELECT count(*) FROM collections;
SELECT count(*) FROM items;
SELECT * FROM collections order by id limit 1;
SELECT * FROM items order by id limit 2;