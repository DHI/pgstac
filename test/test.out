 count 
-------
     1
(1 row)

 count 
-------
   100
(1 row)

    id     |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    content                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     
-----------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 usda-naip | {"id": "usda-naip", "links": [{"rel": "root", "href": "/collection.json", "type": "application/json"}, {"rel": "self", "href": "/collection.json", "type": "application/json"}], "title": "NAIP: National Agriculture Imagery Program", "extent": {"spatial": {"bbox": [[-124.784, 24.744, -66.951, 49.346]]}, "temporal": {"interval": [["2011-01-01T00:00:00Z", "2019-01-01T00:00:00Z"]]}}, "license": "PDDL-1.0", "providers": [{"url": "https://www.fsa.usda.gov/programs-and-services/aerial-photography/imagery-programs/naip-imagery/", "name": "USDA Farm Service Agency", "roles": ["producer", "licensor"]}], "description": "The National Agriculture Imagery Program (NAIP) acquires aerial imagery\nduring the agricultural growing seasons in the continental U.S.\n\nNAIP projects are contracted each year based upon available funding and the\nFSA imagery acquisition cycle. Beginning in 2003, NAIP was acquired on\na 5-year cycle. 2008 was a transition year, and a three-year cycle began\nin 2009.\n\nNAIP imagery is acquired at a one-meter ground sample distance (GSD) with a\nhorizontal accuracy that matches within six meters of photo-identifiable\nground control points, which are used during image inspection.\n\nOlder images were collected using 3 bands (Red, Green, and Blue: RGB), but\nnewer imagery is usually collected with an additional near-infrared band\n(RGBN).", "stac_version": "1.0.0-beta.2", "stac_extensions": []}
(1 row)

                   id                   | stac_version | stac_extensions |                                                                                              geometry                                                                                              |                                                                                                                                                                                                                       properties                                                                                                                                                                                                                       |                                                                                                                                                                                                                                                                                                                                                                                                                                               assets                                                                                                                                                                                                                                                                                                                                                                                                                                               | collection_id |        datetime        
----------------------------------------+--------------+-----------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+---------------+------------------------
 al_m_3008501_ne_16_1_20110815_20111017 | 1.0.0-beta.2 | {eo,projection} | 0103000020E6100000010000000500000008E57D1CCD7755C050C763062AEF3E40F1B913ECBF7755C01152B7B3AF003F4081D07AF8327C55C0D40E7F4DD6003F403FE42D573F7C55C0728C648F50EF3E4008E57D1CCD7755C050C763062AEF3E40 | {"gsd": 1, "datetime": "2011-08-15T00:00:00Z", "naip:year": "2011", "proj:bbox": [601135, 3422859, 607773, 3430446], "proj:epsg": 26916, "providers": [{"url": "https://www.fsa.usda.gov/programs-and-services/aerial-photography/imagery-programs/naip-imagery/", "name": "USDA Farm Service Agency", "roles": ["producer", "licensor"]}], "naip:state": "al", "proj:shape": [7587, 6638], "proj:transform": [1, 0, 601135, 0, -1, 3430446, 0, 0, 1]} | {"image": {"href": "https://naipeuwest.blob.core.windows.net/naip/v002/al/2011/al_100cm_2011/30085/m_3008501_ne_16_1_20110815.tif", "type": "image/tiff; application=geotiff; profile=cloud-optimized", "roles": ["data"], "title": "RGBIR COG tile", "eo:bands": [{"name": "Red", "common_name": "red"}, {"name": "Green", "common_name": "green"}, {"name": "Blue", "common_name": "blue"}, {"name": "NIR", "common_name": "nir", "description": "near-infrared"}]}, "metadata": {"href": "https://naipeuwest.blob.core.windows.net/naip/v002/al/2011/al_fgdc_2011/30085/m_3008501_ne_16_1_20110815.txt", "type": "text/plain", "roles": ["metadata"], "title": "FGDC Metdata"}, "thumbnail": {"href": "https://naipeuwest.blob.core.windows.net/naip/v002/al/2011/al_100cm_2011/30085/m_3008501_ne_16_1_20110815.200.jpg", "type": "image/jpeg", "roles": ["thumbnail"], "title": "Thumbnail"}} | usda-naip     | 2011-08-15 00:00:00+00
 al_m_3008501_nw_16_1_20110815_20111017 | 1.0.0-beta.2 | {eo,projection} | 0103000020E6100000010000000500000050ABE80FCD7B55C0A8AAD0402CEF3E4009E23C9CC07B55C01152B7B3AF003F40698EACFC328055C07C2B1213D4003F403FFED2A23E8055C0728C648F50EF3E4050ABE80FCD7B55C0A8AAD0402CEF3E40 | {"gsd": 1, "datetime": "2011-08-15T00:00:00Z", "naip:year": "2011", "proj:bbox": [595168, 3422804, 601802, 3430387], "proj:epsg": 26916, "providers": [{"url": "https://www.fsa.usda.gov/programs-and-services/aerial-photography/imagery-programs/naip-imagery/", "name": "USDA Farm Service Agency", "roles": ["producer", "licensor"]}], "naip:state": "al", "proj:shape": [7583, 6634], "proj:transform": [1, 0, 595168, 0, -1, 3430387, 0, 0, 1]} | {"image": {"href": "https://naipeuwest.blob.core.windows.net/naip/v002/al/2011/al_100cm_2011/30085/m_3008501_nw_16_1_20110815.tif", "type": "image/tiff; application=geotiff; profile=cloud-optimized", "roles": ["data"], "title": "RGBIR COG tile", "eo:bands": [{"name": "Red", "common_name": "red"}, {"name": "Green", "common_name": "green"}, {"name": "Blue", "common_name": "blue"}, {"name": "NIR", "common_name": "nir", "description": "near-infrared"}]}, "metadata": {"href": "https://naipeuwest.blob.core.windows.net/naip/v002/al/2011/al_fgdc_2011/30085/m_3008501_nw_16_1_20110815.txt", "type": "text/plain", "roles": ["metadata"], "title": "FGDC Metdata"}, "thumbnail": {"href": "https://naipeuwest.blob.core.windows.net/naip/v002/al/2011/al_100cm_2011/30085/m_3008501_nw_16_1_20110815.200.jpg", "type": "image/jpeg", "roles": ["thumbnail"], "title": "Thumbnail"}} | usda-naip     | 2011-08-15 00:00:00+00
(2 rows)

